### ISSUE TITLE
Arrow dataframes do not work yet

### ISSUE THREAD

**Issue by domoritz** [domoritz/streamlit-vega-lite#4]
Somehow, the arrow table doesn't have the right column names so we are transferring the data as json. Maybe we can construct a new arrow table with the right column names in the client.

**Comment by hydrosquall** [comment]
Based on the finding here:

https://github.com/domoritz/streamlit-vega-lite/pull/2#discussion_r486029852

perhaps we could modify `arrow` to consume the whole `Streamlit` `ArrowTable` rather than just an `apache-arrow` `Table`

https://github.com/domoritz/streamlit-vega-lite/blob/master/streamlit-vega-lite/frontend/src/arrow-loader.ts

Then, we could update the following line to get the column name via `aLocalArrowTable.columnTable.getColumnAt(index)` instead of what we currently have.

https://github.com/domoritz/streamlit-vega-lite/blob/6703d2bda46474db7dc3970af91e966824be21a2/streamlit-vega-lite/frontend/src/arrow-loader.ts#L25

**Comment by domoritz** [comment]
@kantuni helped me a bit on the Streamlit Slack. Maybe we get some more help on this. 

> Let me know if there are other helper functions that don’t exist at the moment, but might be helpful. For example getColumns() , getRows(), etc.

**Comment by kantuni** [comment]
ArrowTable has multiple tables for data, rows, columns.

`getCell` combines all of them. Grabbing content from `getCell(rowIndex, columnIndex)` should do the trick.

Try it out, and let me know if you still need my help!

**Comment by domoritz** [comment]
I see. I was hoping to just use the row proxy we use in Vega for normal arrow tables (https://github.com/domoritz/streamlit-vega-lite/blob/master/streamlit-vega-lite/frontend/src/arrow-loader.ts#L9). I guess we can rewrite it to work with the Streamlit arrow table. 

Just wondering, why doesn't Streamlit expose a standard arrow table with column names?

**Comment by kantuni** [comment]
Arrow had an issue converting tables with multiple header rows / columns. To support that use case we decided to split rows, columns, and data into separate tables. We're considering adding more helper functions to the wrapper, so please let us know what you need.

**Comment by domoritz** [comment]
Ahh, makes sense. 

I will try to update the wrapper to support Streamlit `ArrowTable`s but I can see that for most cases where people have a single header table it would be nice to have a method to get a standard Arrow `Table` with column names in it. Besides me, it would also help other libraries that expect an Arrow `Table` such as https://github.com/uwdata/arquero.

**Comment by kantuni** [comment]
> Ahh, makes sense.
> 
> I will try to update the wrapper to support Streamlit `ArrowTable`s but I can see that for most cases where people have a single header table it would be nice to have a method to get a standard Arrow `Table` with column names in it. Besides me, it would also help other libraries that expect an Arrow `Table` such as https://github.com/uwdata/arquero.

I'll mention it during the eng standup.

**Comment by kantuni** [comment]
Hey guys,

I've brought up the issue during the standup, and we decided that we may introduce a method in a wrapper to return the original arrow table. As that might take some time, please let me know if I can help you unblock for the time being.

**Comment by domoritz** [comment]
that would be great. If you could for now tell me how I can get the content of a cell (like, the actual value), that would be great.

**Comment by kantuni** [comment]
> that would be great. If you could for now tell me how I can get the content of a cell (like, the actual value), that would be great.

Have you tried getting `content` from `getCell(rowIndex, columnIndex)`?

**Comment by domoritz** [comment]
Yes, and it's not the content of the cell but some object, null, or empty string. Am I indexing incorrectly?

<img width="765" alt="Screen Shot 2020-09-12 at 17 47 26" src="https://user-images.githubusercontent.com/589034/93007675-10edcd00-f520-11ea-85bf-bfdc226f1ba6.png">

**Comment by domoritz** [comment]
I'm probably indexing incorrectly. What is `columns` and what is `dataColumns`? I guess the first column is an index column from pandas. Is it safe for me to always ignore it or could the first column be a named column?

**Comment by domoritz** [comment]
Ah, I think I got it working by using the column names I get with `getCell(..,...).content`. thank you for your help @kantuni.

**Comment by domoritz** [comment]
Hmm, if I can use the actual index from pandas in my chart, I can easily map back to the data... cool!
